libs/auth/src/openssl.ts

```diff
@@ -434,3 +414,109 @@ export async function createCert(input: CreateCertInput): Promise<Buffer> {
     : [scriptPath, scriptInput];
   return runCommand(command);
 }
+
+//
+// Message signing functions
+//
+
+/**
+ * The input to signMessage. File keys should be in PEM format.
+ */
+export interface SignMessageInput {
+  /**
+   * We use streaming APIs to process the message since the message could be large and infeasible
+   * to store in memory (i.e. in a Buffer), e.g. the message might include the contents of a
+   * multi-GB zip file.
+   *
+   * A seeming alternative would be to require a file path since the openssl command streams under
+   * the hood, but many of our messages will include the contents of a file preceded by a custom
+   * prefix, so there is no file path to provide (unless a temporary file is created, which too
+   * would have to be done with streaming APIs).
+   */
+  message: Stream;
```

Comment:
I did some manual testing with a dummy 2GB file. Creating its VVSG2 data protection signature file took on average between 3 to 5 seconds on my machine with the streaming approach that I've landed on. Earlier iterations without streaming caused my JS heap to run out of memory. Good validation!